src/app.rs

```diff
@@ -1025,17 +1188,208 @@ impl eframe::App for ProfApp {
                 windows.push(Window::new(extra, index));
                 let window = windows.last_mut().unwrap();
                 cx.total_interval = cx.total_interval.union(window.config.interval);
-                cx.view_interval = cx.total_interval;
+                ProfApp::zoom(cx, cx.total_interval);
+            }
+
+            let visual = if cx.viewing_mode {
+                egui::Visuals::light()
+            } else {
+                egui::Visuals::dark()
+            };
+
+            // swap to dark mode
```

Comment:
The code seems clear to me, I don't think we need this comment.

But I would like a label in the UI, because as a user the interface is not obvious.